flake.nix

```diff
@@ -0,0 +1,52 @@
+{
+  description = "A flake for s2n-tls";
+
+  inputs.nixpkgs.url = github:NixOS/nixpkgs/nixos-22.11;
+
+  outputs = { self, nix, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let pkgs = nixpkgs.legacyPackages.${system};
+      in rec {
+        packages.s2n-tls = pkgs.stdenv.mkDerivation {
+          src = self;
+          name = "s2n-tls";
+          inherit system;
+          # TODO: override the s2n-tls version
+
+          buildInputs = [
+            pkgs.cmake
+            pkgs.openssl_3
```

Comment:
Do we want this as the default? That means if we make a mistake in the definitions below `packages.s2n-tls-openssl11` it will fall back to this rather than error?

Reply:
We need something for the default build or it will fail.  We are putting in a libcrypto check and we can pose how this should be handled to the nix community.